### PR TITLE
 testharness: collect and print parser errors better

### LINKED ISSUE #10005: Improve error message of HIT parser warnings in TestHarness
## Rationale

The new HIT parser is more strict on the correctness of the input file syntax. This is causing several developers to encounter the "FATAL PARSER ERROR" message from the TestHarness. Unfortunately, the actual error is difficult to find since it's not colored and interleaved with normal TestHarness output. The parser messages should come out in a summary at the bottom. It wouldn't hurt to have the message colored as well in the TestHarness status.

## Description

Add a syntax error to a test spec file (e.g. tests) and run the TestHarness to reproduce. 

## Impact

Easier for developers to find errors, no impact to functionality.

### PR DESCRIPTION
**Pull request by rwcarlsen**
fixes #10005

### CODE REVIEW

File: test/tests/kernels/simple_diffusion/tests [hunk 1]
```diff
@@ -1,6 +1,8 @@
 [Tests]
   [./test]
     type = 'Exodiff'
+    type = 'Exodiff'
```
Comment (permcody) [inline test/tests/kernels/simple_diffusion/tests]: This is good, but should be a TestHarness test!

**Comment by rwcarlsen** [comment]
Where are we sitting with this now?

File: python/FactorySystem/InputParameters.py [hunk 2]
```diff
@@ -45,6 +45,11 @@ def isValid(self, name):
     def __contains__(self, item):
         return item in self.desc
 
+    def get(self, key, default=None):
```
Comment (permcody) [inline python/FactorySystem/InputParameters.py]: While this is useful. I've been trying to keep the interfaces between MOOSE and this class the same just os users can move between them and have the same interface. There isn't a method like this on the MOOSE side.

Comment (rwcarlsen) [reply python/FactorySystem/InputParameters.py]: Input parameters object implements the python dictionary operator (``[bla]`` indexing).  This method just makes that interface consistent with the python dictionary interface which also has a ``.get`` with the same semantics.  But I don't feel that strongly about it either way.

Comment (permcody) [reply python/FactorySystem/InputParameters.py]: Let's remove it. It's just a convenience and isn't necessary. The rest of this looks fine. I'll review and merge after that.
Thanks

Comment (rwcarlsen) [reply python/FactorySystem/InputParameters.py]: done.

**Comment by rwcarlsen** [comment]
This is ready now.   Comments should all be addressed.

**Review by permcody — APPROVED** [review]

**Pull request merged**